chia/wallet/util/address_type.py

```diff
@@ -0,0 +1,71 @@
+from enum import Enum
+from functools import wraps
+from typing import Any, Callable, Optional, Set, Type, TypeVar
+
+from chia.util.bech32m import bech32_decode
+from chia.util.config import selected_network_address_prefix
+
+_T_CurrentNetworkAddressPrefix = TypeVar("_T_CurrentNetworkAddressPrefix", bound="CurrentNetworkAddressPrefix")
+
+
+# Singleton representing the current network address prefix. Can be updated in case the config changes.
+# Since AddressType is an Enum, it's not possible to have a class attribute within AddressType that is
+# mutable. To support the case where the selected network address prefix changes, we use this class
+# to store that prefix.
+class CurrentNetworkAddressPrefix:
+    current: str = selected_network_address_prefix()
+
+    @classmethod
+    def update(cls: Type[_T_CurrentNetworkAddressPrefix], value: str) -> None:
+        cls.current = value
+
+
+class AddressType(Enum):
+    XCH = "xch"
+    TXCH = "txch"
+    NFT = "nft"
+    DID = "did:chia:"
```

Comment:
The issue may more be that it should be tnft and tdid...

Reply:
Perhaps.  I would still suggest that those are perhaps not address types but rather an orthogonal dimension.  We are unlikely to hint function `f()` to handle `tnft` and `g()` to handle `nft`.

Reply:
Yes, we should consider making testnet variations for DID and NFT identifiers.

I'll think a bit more about the testnet/mainnet issue. It is a goal to prevent cross-chain address usage (mainnet can't send to TXCH, testnet can't send to XCH), but I take the point about when would we ever use AddressType.TXCH (basically never).